src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2385,6 +2386,29 @@ void TorrentImpl::setMetadata(const TorrentInfo &torrentInfo)
     });
 }
 
+void TorrentImpl::setSslCertificate(const QByteArray &certificate, const QByteArray &privateKey, const QByteArray &dhParams)
+{
+    m_nativeHandle.set_ssl_certificate_buffer(certificate.toStdString(), privateKey.toStdString(), dhParams.toStdString());
+
+    const Path baseDir = m_session->sslCertificatesDirectory();
+    const auto [certPath, keyPath, dhPath] = m_session->sslCertificatesPathsForTorrent(id());
+
+    if (!baseDir.exists())
+    {
+        Utils::Fs::mkpath(baseDir);
+        QFile(keyPath.data()).setPermissions(QFile::ReadOwner | QFile::WriteOwner | QFile::ExeOwner | QFile::ReadUser | QFile::WriteUser | QFile::ExeUser);
```

Comment:
Why is it here? Shouldn't this be done after you've created the file? Besides, it's there too...
Or did you intend to apply it to `baseDir` here instead of `keyPath`?

Reply:
yes, I intended to use baseDir